packages/shared/lib/services/sync/manager.service.ts

```diff
@@ -432,13 +435,27 @@ export class SyncManagerService {
         }
         const schedule = schedules.value.get(sync.id);
         let frequency = sync.frequency;
+        const syncConfig = await getSyncConfigByParams(environmentId, sync.name, providerConfigKey);
         if (!frequency) {
-            const syncConfig = await getSyncConfigByParams(environmentId, sync.name, providerConfigKey);
             frequency = syncConfig?.runs || null;
         }
         if (!schedule) {
             throw new Error(`Schedule for sync ${sync.id} and environment ${environmentId} not found`);
         }
+
+        const countRes = await recordsService.getRecordCountsByModel({ connectionId: sync.nango_connection_id, environmentId });
+        if (countRes.isErr()) {
+            logger.error(`Failed to get record count for sync ${sync.id} in environment ${environmentId}: ${stringifyError(countRes.error)}`);
```

Comment:
do we want this to be optional? Some customers might build logic around those numbers and if they goes to zero suddenly could be problematic